### ISSUE TITLE
Non constant fur length

### ISSUE THREAD

**Issue by br3eze** [Arnklit/ShellFurGodot#6]
**Describe the bug**
Length is not so big, so should they be so long?

**To Reproduce**
Fur + Wind (settings are on the screenshot)

**Expected behavior**
Constant long

**Screenshots**
![Screenshot_20220309_155649](https://user-images.githubusercontent.com/5236855/157408397-44b11544-7e1e-487e-acc9-d7053d71ee08.png)

**Environment**
Arch Linux
Intel hd 620 / rtx2060
Godot 3.4.x
Plugin 0.2.0
GLES3

**Comment by Arnklit** [comment]
Yeah you are right. Wind can effect the length of the fur. It's a bit difficult to avoid without making the vertex shader way more complicated, but I think I can improve it quite a bit. I'm getting started on some improvements for the next version right now and I'll be sure to improve this.

**Comment by br3eze** [comment]
May be it will be helpful. It is not about only a wind, also it ruins all when you use high gravity or too low dump values.

**Comment by br3eze** [comment]
Hello. We are preparing to release and want to use you plugin.
Should we wait or search another way to deal with a fur?

**Comment by Arnklit** [comment]
Hi Breeze, I've been on summer vacation, so sorry for the delay. Is this in relation to this bug? Or just the plugin in general?

**Comment by br3eze** [comment]
It's ok, no hurry. And if we can set length and spring as gradients with a fur texture, then only this bug.

Why spring? Because there is no way to make an adequate fur on a low-poly model.
Just for example: 10 cm long fur on a dog 1500 polys model. It will lay on the model and will show every polygon on the dogs back. I have no clue how to make it right.

**Comment by br3eze** [comment]
"And if we can set length and spring as gradients with a fur texture" - now I know that we can't.

**Comment by Arnklit** [comment]
I'm sorry br3eze I'm really not sure what your limitations are and what you need, could you try and post some screenshots of the issues? I know there are issues with fur not keeping it's length with strong physics forces, but that's not something I can easily fix.

I'm currently spending my time updateing my other plugin Waterways to Godot 4.0, but when I'm done with that I hope to get to working on updating shell fur to Godot 4.0.

**Comment by br3eze** [comment]
Yeah, may be I was not quite concrete or it is my English. This is what I mean:
![image_2022-07-19_23-51-02](https://user-images.githubusercontent.com/5236855/206921547-44909584-fa12-45d6-865d-c9bf452a4151.png)

You can't make a gradient of a fur length: short on feet and ears, and long on a body. Just an example for clearness.

Also, I don't know how to orient a nap - it just grows perpendicularly. So, your personage looks very weird.

I know nothing about shaders creation yet. But may be an alpha rgb channels values of a texture can bu used for some parameters.
How and can it be even done in a shader code not even imagine. Just sharing a thought.

**Comment by Arnklit** [comment]
You can use the ldtg texture to control the length. If all you want to control is the length, you should put your length data in the red channel and leave the other channels full white. The  examples in the demo scene also use this, so you can look there.

This is an example texture, that has a gradient in the red channel, but is white in the other channels, try that out as the ldtg texture.
![ldgt](https://user-images.githubusercontent.com/4955051/207009463-e26a0c79-c733-48ae-960b-fc5df82530e3.png)

If you want to orient the growth you need to make a blend shape in Blender and set that in the "Blendshape Styling" section. I also do this in the demo, the Rat has a blend shape that orients the fur, it's also detailed in the documentation.

![](https://user-images.githubusercontent.com/4955051/111274654-2bb71080-862d-11eb-8696-60dab4515434.gif)

**Comment by br3eze** [comment]
Please excuse my inattention and thank You for Your patient!

**Comment by br3eze** [comment]
Unusable in real projects, if Your personage should move. If it can move fast, unusable^2.
What can we do with this? How to clamp fur's length?
I mean, may be You know how to reduce this effect at least?

https://dropmefiles.com/iBtId
